e2e/test.mjs

```diff
@@ -0,0 +1,76 @@
+import { NodeSDK } from '@opentelemetry/sdk-node';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-proto';
+import { OTLPMetricExporter } from '@opentelemetry/exporter-metrics-otlp-proto';
+import { PeriodicExportingMetricReader } from '@opentelemetry/sdk-metrics';
+import { OTLPLogExporter } from '@opentelemetry/exporter-logs-otlp-proto';
+import { LoggerProvider, SimpleLogRecordProcessor } from '@opentelemetry/sdk-logs';
+import { SimpleSpanProcessor } from '@opentelemetry/sdk-trace-base';
+import { diag, DiagConsoleLogger, DiagLogLevel, trace, metrics } from '@opentelemetry/api';
+import { logs } from '@opentelemetry/api-logs';
+
+// Enable diagnostic logging (optional)
+diag.setLogger(new DiagConsoleLogger(), DiagLogLevel.INFO);
+
+const collectorUrl = 'http://localhost:4318/v1';
+
+// Set up trace exporter with SimpleSpanProcessor
+const traceExporter = new OTLPTraceExporter({
+    url: `${collectorUrl}/traces`,
+});
+const spanProcessors = [new SimpleSpanProcessor(traceExporter)];
+
+// Set up metric exporter
+const metricExporter = new OTLPMetricExporter({
+    url: `${collectorUrl}/metrics`,
+});
+const metricReader = new PeriodicExportingMetricReader({
+    exporter: metricExporter,
+    exportIntervalMillis: 1000,
+    exportTimeoutMillis: 1000,
+});
+
+// Set up log exporter
+const logExporter = new OTLPLogExporter({
+    url: `${collectorUrl}/logs`,
+});
+const loggerProvider = new LoggerProvider({
+    processors: [new SimpleLogRecordProcessor(logExporter)],
+});
+
+// Set up OpenTelemetry SDK
+const sdk = new NodeSDK({
+    spanProcessors,
+    metricReader,
+    loggerProvider,
```

Comment:
issue: `loggerProvider` is not an option that can be passed to `NodeSDK` AFAIK, only `logRecordProcessors` is.

I am surprised that this passes the tests, could it be that the tests are not failing even though they should 🤔

Reply:
I looked into it in a bit more detail as this left me scratching my head.
Looks like in `NodeSDK` when there's nothing defined we default to enabling the Logs SDK - and the default setup just happened to point towards the collector, which made the tests pass.